### PR TITLE
PR suggestions

### PR DESCRIPTION
**Pull request by Itamaram**

### CODE REVIEW

File: Xero.NetStandard.OAuth2/Client/ApiException.cs [hunk 1]
```diff
@@ -33,13 +33,13 @@ public class ApiException : Exception
         /// Gets or sets Retry-After value
         /// </summary>
         /// <value>The Retry-After (Http response header Retry-After).</value>
-        public int? RetryAfter { get; set; }
+        public int? RetryAfter { get; private set; }
```
Comment (Itamaram) [inline Xero.NetStandard.OAuth2/Client/ApiException.cs:36]: I am not sure if having a public set is part of convention or auto generation, but for correctness it should be private.

File: Xero.NetStandard.OAuth2/Client/ApiException.cs [hunk 2]
```diff
@@ -33,13 +33,13 @@ public class ApiException : Exception
         /// Gets or sets Retry-After value
         /// </summary>
         /// <value>The Retry-After (Http response header Retry-After).</value>
```
Comment (Itamaram) [inline Xero.NetStandard.OAuth2/Client/ApiException.cs:35]: I saw you removed my added comment regarding the actual value (eg that it is measured in minutes). As a consumer of this SDK, this saves me having to look this up online. I would strongly recommend reinstating it.

File: Xero.NetStandard.OAuth2/Client/ApiException.cs [hunk 3]
```diff
@@ -33,13 +33,13 @@ public class ApiException : Exception
...
-        public int? RetryAfter { get; set; }
+        public int? RetryAfter { get; private set; }
 
         /// <summary>
         /// Gets or sets the Limit Type value
         /// </summary>
         /// <value>The Limit Type (Http response header X-Rate-Limit-Problem).</value>
-        public string LimitType { get; set; }
+        public string LimitType { get; private set; }
```
Comment (Itamaram) [inline Xero.NetStandard.OAuth2/Client/ApiException.cs:42]: As above regarding private. Also as above regarding the possible values of this variable included in the xml comment

File: Xero.NetStandard.OAuth2/Client/ApiException.cs [hunk 4]
```diff
@@ -69,14 +69,15 @@ public ApiException(int errorCode, string message, dynamic errorContent = null)
...
-        /// Initializes a new instance of the <see cref="ApiException"/> class.
+        /// Initializes a new instance of the <see cref="ApiException"/> class
+        /// for the case when a rate limit is hit
         /// </summary>
         /// <param name="errorCode">HTTP status code.</param>
         /// <param name="message">Error message.</param>
         /// <param name="errorContent">Error content.</param>
-        /// <param name="limitType">Error content.</param>
-        /// <param name="retryAfter">Error content.</param>
-        public ApiException(int errorCode, string message, int retryAfter, dynamic errorContent = null, string limitType = null) : base(message)
```
Comment (Itamaram) [inline Xero.NetStandard.OAuth2/Client/ApiException.cs:79]: None of these constructor arguments are optional for this constructor

Comment (Itamaram) [inline Xero.NetStandard.OAuth2/Client/ApiException.cs:79]: As a stylistic choice, I would argue against sandwiching the `errorContent` argument between two new ones. Keep it in the same absolute position as the other ctor (ie after `message`), or the same relative position (ie last)

File: Xero.NetStandard.OAuth2/Client/ApiException.cs [hunk 5]
```diff
@@ -69,14 +69,15 @@ public ApiException(int errorCode, string message, dynamic errorContent = null)
...
+        /// <param name="limitType">The value of the HTTP Header <c>X-Rate-Limit-Problem</c></param>
```
Comment (Itamaram) [inline Xero.NetStandard.OAuth2/Client/ApiException.cs:78]: While the content of this `param` comment can be anything, I would argue it should be set to something which is not copy pasted from the line above it.

File: Xero.NetStandard.OAuth2/Client/Configuration.cs [hunk 6]
```diff
@@ -59,7 +59,7 @@ public class Configuration : IReadableConfiguration
                     int ra = response.Headers.TryGetValue("Retry-After", out var ras) && int.TryParse(ras.FirstOrDefault(), out ra)
                         ? ra
                         : 0;
-                    return new ApiException(status, string.Format("Xero API {0} rate limit error calling {1}. Retry-After {2}", limitType, methodName, ra), ra, response.Content, limitType);
+                    return new ApiException(status, string.Format("Xero API {0} rate limit error calling {1}", limitType, methodName), response.Content, ra, limitType);
```
Comment (Itamaram) [inline Xero.NetStandard.OAuth2/Client/Configuration.cs:62]: This is actually absolutely **critical**.

The current advice provided by Xero regarding capturing retry errors, is to check the existing Exception's message against a regex. If you change the error message, you lose all backwards compatibility.

**Comment by RettBehrens** [comment]
Resolved with the latest release [3.22.1](https://github.com/XeroAPI/Xero-NetStandard/releases/tag/3.22.1)

**Pull request closed without merging**